tests/services/apiService.test.js

```diff
@@ -40,3 +40,65 @@ describe("fetchQuestCategoryData function", () => {
     expect(result).toEqual(mockResponse);
   });
 });
+describe("getQuestActivityData function", () => {
+  beforeEach(() => {
+    fetch.mockClear();
+  });
+  it("should fetch and return data for a valid id", async () => {
+    //mock data for successful response
+    const mockData = [{ date: "2024-04-04 04", participants: 2 }];
+    fetch.mockResolvedValueOnce({
+      json: () => Promise.resolve(mockData),
+    });
+    const result = await getQuestActivityData(1);
+    expect(fetch).toHaveBeenCalledWith(
+      `${API_URL}/analytics/get_quest_activity?id=1`
+    );
+    expect(result).toEqual(mockData);
+  });
+  it("should handle API returning no response", async () => {
+    // Mock fetch response with no data
+    const mockResponse = undefined;
+    fetch.mockResolvedValueOnce({ json: () => Promise.resolve(mockResponse) });
+
+    const result = await getQuestActivityData("invalidId");
+    expect(fetch).toHaveBeenCalledWith(
+      `${API_URL}/analytics/get_quest_activity?id=invalidId`
+    );
+    expect(result).toBeUndefined();
+  });
+
+  it("should handle unexpected response format", async () => {
+    // Mock fetch response with unexpected data format
+    const mockResponse = "Unexpected response format";
```

Comment:
you can try setting the response as an empty array in this since that is the unexpected format here. Sending something like id=70 should help you see this case

Reply:
Okay
I will get on that asap